### PR TITLE
A0-1340: add info level logs to finality-aleph components

### PR DESCRIPTION
**Pull request by maciejnems**
# Description

This PR adds new logs for finality-aleph components. Specificaly to:

**Network status report**
Example report for 2 connected authorities plus 2 connected archivists:
```
2022-09-12 16:02:59 Network status report: validator connected peers - 2 [12D3KooWCmz8XwEief5Z9Dt42MqzSJcjzD48wEoNGXxBmnodAv5S, 12D3KooWQ49SdyE3fM3vw8VupK5GvzGfvVkJMoM9UeaUQD5FvTAX]; generic connected peers - 4;
```

**Connection Manager Service status report** 
Example report for 1 missing undetected authority:
```
2022-09-12 16:02:59 Connection Manager Service status report: authenticated authorities: authenticated authorities: 1: 3/4 {2: 12D3KooWQ49SdyE3fM3vw8VupK5GvzGfvVkJMoM9UeaUQD5FvTAX, 3: 12D3KooWCmz8XwEief5Z9Dt42MqzSJcjzD48wEoNGXxBmnodAv5S}, 0: 3/4 {3: 12D3KooWCmz8XwEief5Z9Dt42MqzSJcjzD48wEoNGXxBmnodAv5S, 2: 12D3KooWQ49SdyE3fM3vw8VupK5GvzGfvVkJMoM9UeaUQD5FvTAX}; missing authorities: 1: [0], 0: [0];
```
Example report of all authorities authenticated:
```
2022-09-12 16:18:13 Connection Manager Service status report: authenticated authorities: 1: 4/4 {2: 12D3KooWQ49SdyE3fM3vw8VupK5GvzGfvVkJMoM9UeaUQD5FvTAX, 1: 12D3KooWSPrwNJc6WM7F9MTEtBfy5k2VrTgYkxUFuyNimjJmhM9h, 3: 12D3KooWCmz8XwEief5Z9Dt42MqzSJcjzD48wEoNGXxBmnodAv5S}, 0: 4/4 {1: 12D3KooWSPrwNJc6WM7F9MTEtBfy5k2VrTgYkxUFuyNimjJmhM9h, 3: 12D3KooWCmz8XwEief5Z9Dt42MqzSJcjzD48wEoNGXxBmnodAv5S, 2: 12D3KooWQ49SdyE3fM3vw8VupK5GvzGfvVkJMoM9UeaUQD5FvTAX};
```

**Block Signature Aggregator status report**
Example report:
```
Block Signature Aggregator status report: started hashes - 7; collected sginatures - 0; hashes in queue - 0;
```
Example report when hash queue is not empty:
```
2022-09-12 16:24:38 Block Signature Aggregator status report: started hashes - 7; collected sginatures - 0; hashes in queue - 7; front of hash queue - 0xd5b8…ee85 for - 2.144170386s;
```

**Justification request queue**
When clearing justification request queue, log about it was changed from debug to info log level,

## Type of change

- Add info logs

### CODE REVIEW

File: finality-aleph/src/aggregation/aggregator.rs [hunk 1]
```diff
@@ -75,6 +82,32 @@ impl<H: Copy + Hash, PMS> BlockSignatureAggregator<H, PMS> {
...
+    pub fn status_report(&self) {
+        let mut status = String::from("Block Signature Aggregator status report: ");
+
+        status.push_str(&format!(
+            "started hashes - {:?}; ",
+            self.started_hashes.len()
+        ));
+
+        status.push_str(&format!(
+            "collected sginatures - {:?}; ",
```
Comment (DamianStraszak) [inline finality-aleph/src/aggregation/aggregator.rs:95]: typo `sginatures`

File: finality-aleph/src/aggregation/aggregator.rs [hunk 2]
```diff
@@ -75,6 +82,32 @@ impl<H: Copy + Hash, PMS> BlockSignatureAggregator<H, PMS> {
...
+            self.signatures.len()
+        ));
+
+        status.push_str(&format!("hashes in queue - {:?}; ", self.hash_queue.len()));
+
+        if let Some(hash) = self.hash_queue.front() {
+            status.push_str(&format!(
+                "front of hash queue - {} for - {:?}; ",
+                hash,
+                Instant::now() - self.front_instant
```
Comment (DamianStraszak) [inline finality-aleph/src/aggregation/aggregator.rs:105]: This code might panic -- avoid unchecked arithmetic on instants, because monotonicity might fail. Use `saturating_duration_since().as_f64`.  Please, print it with precision of `2` decimal places.

File: finality-aleph/src/aggregation/aggregator.rs [hunk 3]
```diff
@@ -41,6 +43,7 @@ impl<H: Copy + Hash, PMS> BlockSignatureAggregator<H, PMS> {
             signatures: HashMap::new(),
             hash_queue: VecDeque::new(),
             started_hashes: HashSet::new(),
+            front_instant: Instant::now(),
```
Comment (DamianStraszak) [inline finality-aleph/src/aggregation/aggregator.rs:46]: Maybe better `last_change` or something suggesting what this stands for?

File: finality-aleph/src/substrate_network.rs [hunk 4]
```diff
@@ -69,6 +69,12 @@ impl Decode for PeerId {
     }
 }
 
+impl fmt::Display for PeerId {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
```
Comment (DamianStraszak) [inline finality-aleph/src/substrate_network.rs:74]: I was thinking maybe to print this in the form of `prefix...sufix`, where both prefix and sufix are, say `8` characters only. Because currently this is quite verbose. 

@timorl what do you think?

File: finality-aleph/src/justification/requester.rs [hunk 5]
```diff
@@ -128,7 +128,7 @@ where
                 }
             }
             SchedulerActions::ClearQueue => {
-                debug!(target: "aleph-justification", "Clearing queue");
+                info!(target: "aleph-justification", "Clearing justification request queue");
```
Comment (DamianStraszak) [inline finality-aleph/src/justification/requester.rs:131]: This is not a good message to have on level `info`. It doesn't give any useful information about the internal state of the component. What we would like to catch here at least, is that we keep requesting the same block, and fail for a prolonged period of time. In such a case we want to learn the number and hash of the requested block.

**Review by DamianStraszak — COMMENTED** [review]
I have left some comments for everything except network-related logs.

I don't know enough about the inner workings of network to give useful feedback about the relevant logs here -- @timorl should be a better person to ask. I will be looking at the network code for some time on Tuesday, so I might be able to give some comments later. For now, a little comment I can give is that it is useful to display list of peers, etc. sorted by `NodeIndex`, not in arbitrary order.

File: finality-aleph/src/substrate_network.rs (hunk 4 above)
Comment (timorl) [reply finality-aleph/src/substrate_network.rs:74]: Technically this would usually be enough to find the related PeerId in other logs, but I'm worried that using it for different purposes would be harder (e.g. iirc there is an RPC call to mark a node as reserved and it requires the whole PeerId). So weak (since the alternative uses are unlikely) preference for the full thing being printed, but if you think this hurts the logs too much I'm fine with printing just the version you suggested.

File: finality-aleph/src/lib.rs [hunk 6]
```diff
@@ -38,6 +38,7 @@ mod nodes;
 mod party;
 mod session;
 mod session_map;
+pub mod status;
```
Comment (timorl) [inline finality-aleph/src/lib.rs:41]: I'm not convinced this is necessary as a separate module, since it just very thinly pack a `ticker`. Perhaps just move the `STATUS_TICKER_DELAY` (maybe rename to `STATUS_REPORT_INTERVAL`?) to this `lib.rs` file and crate plain tickers everywhere?

If not, then at least remove the `pub` here, it's unnecessary.

**Review by timorl — APPROVED** [review]
LGTM, just one comment on top of Damian's.

File: finality-aleph/src/network/manager/service.rs [hunk 7]
```diff
@@ -544,6 +544,77 @@ impl<NI: NetworkIdentity, D: Data> Service<NI, D> {
             }
         }
     }
+
+    pub fn status_report(&self) {
+        let mut status = String::from("Connection Manager Service status report: ");
```
Comment (DamianStraszak) [inline finality-aleph/src/network/manager/service.rs:549]: Let's shorten this to "Connection Manager status:"

**Comment by timorl** [comment]
Ah, one additional request – could you bump the two first (not the last one) logs in `network/service.rs` in the `peer_sender` function to `info`? It turns out they don't really happen in usual operation and they should be a symptom of something going quite wrong usually.

File: finality-aleph/src/substrate_network.rs (hunk 4 above)
Comment (maciejnems) [reply finality-aleph/src/substrate_network.rs:74]: I guess the first `8` characters are always the same in PeerId. So maybe we could print prefix of `4` and sufix of `8`?

File: finality-aleph/src/lib.rs (hunk 6 above)
Comment (maciejnems) [reply finality-aleph/src/lib.rs]: removed it

**Review by timorl — APPROVED** [review]
Some nits.

File: finality-aleph/src/network/service.rs [hunk 8]
```diff
@@ -112,13 +116,13 @@ impl<N: Network, D: Data> Service<N, D> {
                         match network.sender(peer_id, protocol) {
                             Ok(sender) => senders.entry(protocol).or_insert(sender),
                             Err(e) => {
-                                debug!(target: "aleph-network", "Failed creating sender. Dropping message: {:?}", e);
+                                info!(target: "aleph-network", "Failed creating sender. Dropping message: {}", e);
```
Comment (DamianStraszak) [inline finality-aleph/src/network/service.rs:119]: Are we sure this will not create spam? I definitely want to avoid a situation where the log is spammed with this. 
Each type of message should be upper bounded to appear once per 10s or some such. 

Similarly for `Failed sending data to peer`.

Just create some aggregate statistics on failures and print them every 10 sec.

Comment (maciejnems) [reply finality-aleph/src/network/service.rs:119]: After some research this will only log in case `StreamClosed` notification did not happen before trying to send here... or if there is some strange bug in substrate. So seems like this log might be undeterministic ://  Also after some thinking this does not sound like info log. Aggregation could be the way to go.

Comment (maciejnems) [reply finality-aleph/src/network/service.rs:119]: There is one problem though. This function is called per peer, so this is not that easy to aggregate statistics from all those at one place. I don't think sending such data over channels is worth it. That said we could add this aggregation to `send_to_peer` function. Everything that is sent to senders goes through this function. I think there we can aggregate each failure to send.

Still, I'm wondering, what exactly should we aggregate:
*  Should we aggregate number of unsent messages per peer? (I guess there might be a lot of peers missing, and this could cause a lot of spam) 
* Should we aggregate just what peers we failed to send? (same nit against as in above point)
* Just aggregating number of unsent messages is not that useful. 

And also for how long should we aggregate them? Should we print peers/messages lost since last status report? I think something like this would be the way to go, but 10 seconds seems like a really short time to reset such aggregation.

So to sum up, I would personally do this aggregation in `send_to_peer`, aggregate unsent messages per user and reset it once every 5-10 status reports?

File: finality-aleph/src/justification/requester.rs (hunk 5 above)
Comment (maciejnems) [reply finality-aleph/src/justification/requester.rs:131]: I guess we could save last num that we requested, save how many times and save a block hash associated with this num. Status report could be called every 10 seconds in component above.

I implemented this in a way, that if there is no pending status report, it should log
```
Justification requester status report: tries - 0;
```
and otherwise, something like that
```
Justification requester status report: tries - 1; requested block number - 5; hash - 0x5d6d…614a;
```

File: finality-aleph/src/network/service.rs (hunk 8 above)
Comment (timorl) [reply finality-aleph/src/network/service.rs:119]: For context, I suggested this change because there were _no_ such logs (at the appropriate level) even when there were problems with connections – as far as I can tell, these are really rare. Sending failures usually turn up in a completely different part of the code.

On the other hand maybe this is not good enough reasoning for level bumping, so perhaps we shouldn't be doing that. I'm kinda skeptical the setup needed to aggregate failure logs here is worth it, especially since, as @maciejnems explained it would have to add some centralization and cross-thread communication, which is usually more complex than one would hope for in logging code.

File: finality-aleph/src/justification/requester.rs [hunk 9]
```diff
@@ -16,6 +16,64 @@ use crate::{
...
+            self.block_hash = None;
+            self.tries = 1;
+        }
+    }
+
+    fn insert_hash(&mut self, hash: B::Hash) {
+        self.block_hash = Some(hash);
+    }
+
+    fn try_reset(&mut self, number: NumberFor<B>) {
```
Comment (DamianStraszak) [inline finality-aleph/src/justification/requester.rs:49]: What's the purpose of this reset?

File: finality-aleph/src/justification/requester.rs [hunk 10]
```diff
@@ -16,6 +16,64 @@ use crate::{
     network, Metrics,
 };
 
+/// This structure is created for keeping and reporting status of BlockRequester
+pub struct LastRequest<B: BlockT> {
```
Comment (DamianStraszak) [inline finality-aleph/src/justification/requester.rs:20]: Why the strange name? Shouldn't it be called something like `JustificationRequestStatus`?

File: finality-aleph/src/justification/requester.rs (hunk 5 above)
Comment (DamianStraszak) [reply finality-aleph/src/justification/requester.rs]: Thanks. This looks good now, but could you also make sure the default (i.e. healthy node) is printing nothing, instead of printing some generic message? This could be done by printing only if `tries` crosses some threshold.

File: finality-aleph/src/network/service.rs (hunk 8 above)
Comment (DamianStraszak) [reply finality-aleph/src/network/service.rs:119]: I cannot really provide any useful insight here, on top of what @timorl already said. Aggregation per peer does not make much sense to me, as it does not guarantee satisfaction of the no-spam condition. 

If the only satisfying solution is complex, maybe you should just move it out of scope of this PR, and create a new task for it?

File: finality-aleph/src/justification/requester.rs (hunk 9 above)
Comment (maciejnems) [reply finality-aleph/src/justification/requester.rs:49]: So that we don't log for example 3 retries for block after we received it, but before we request another block number.

File: finality-aleph/src/justification/requester.rs (hunk 5 above)
Comment (maciejnems) [reply finality-aleph/src/justification/requester.rs]: Is 3 tries a good threshold?

File: finality-aleph/src/justification/requester.rs (hunk 10 above)
Comment (maciejnems) [reply finality-aleph/src/justification/requester.rs:20]: It was supposed to be only status of LastRequest, but `JustificationRequestStatus` sounds like a better name.

File: finality-aleph/src/justification/requester.rs (hunk 9 above)
Comment (DamianStraszak) [reply finality-aleph/src/justification/requester.rs:49]: But why would that be wrong? 
We are logging info about this particular component, and don't care what happened outside. Actually the logs are here mostly because there might be bugs when interacting with outside components. In particular, if we are logging info about requests, let's focus on requests. Whether we received something is a different matter.

Comment (maciejnems) [reply finality-aleph/src/justification/requester.rs:49]: Ok, good point. I will remove it then.

File: finality-aleph/src/justification/requester.rs (hunk 5 above)
Comment (DamianStraszak) [reply finality-aleph/src/justification/requester.rs]: Hard to say -- this would be something to figure out. Does it happen to have 3 tries if everything is healthy? What's the typical behavior of this component when things are healthy?

Alternatively, if you don't want to play with that, it's probably fine to always print, but with lower frequency, something like 60 sec should be fine. Also generally I would recommend making separate constants, not share the 10s one between all. It might make sense to increase/reduce them later individually.

Comment (maciejnems) [reply finality-aleph/src/justification/requester.rs]: When things are healthy nothing should be requested more then once.

Typical behavior is such that after three retries without any finalization in mean time, justification queue is cleared. So usually if this is 3 we know, that something might be going on, and we cleared the queue to fix that. 

On topic of frequency, actually 10 seconds might be too frequently. Some of the logs we have here are quite big. Maybe we could just change all of them to 20/30 seconds

Comment (DamianStraszak) [reply finality-aleph/src/justification/requester.rs]: Yes, you can change others to 20-30 seconds.

And regarding this particular one -- you can make it print only if, say, `tries>=2`. Alternatively, just make it rare, like 60sec, and you can ten print every time.

File: finality-aleph/src/network/service.rs (hunk 8 above)
Comment (timorl) [reply finality-aleph/src/network/service.rs:119]: Oh yeah, this specific change will become mostly irrelevant with the separate validator network, duh. >.<

Comment (DamianStraszak) [reply finality-aleph/src/network/service.rs]: Ok, then if you think this will not create much spam, you can leave it on `info`, otherwise better to change to `debug` and leave it for the future.

Comment (maciejnems) [reply finality-aleph/src/network/service.rs]: I changed it back to debug

**Review by DamianStraszak — APPROVED** [review]
